java/ddlogapi/DDlogAPI.java

```diff
@@ -412,4 +410,107 @@ static public synchronized void logSetCallback(int module, ObjIntConsumer<String
             logCBInfo.put(module, new_cbinfo);
         }
     }
+
+    /**
+     * Run an external process by executing the specified command.
+     * @param commands        Command and arguments.
+     * @param workdirectory   If not null the working directory.
+     * @return                The exit code of the process.  On error prints
+     *                        the process stderr on stderr.
+     */
+    static int runProcess(List<String> commands, String workdirectory) {
+        try {
+            ProcessBuilder pb = new ProcessBuilder(commands);
+            pb.redirectErrorStream(true);
+            if (workdirectory != null) {
+                pb.directory(new File(workdirectory));
+            }
+            Process process = pb.start();
+
+            StringBuilder out = new StringBuilder();
+            BufferedReader br = new BufferedReader(new InputStreamReader(process.getErrorStream()));
+            String line = null;
+            while ((line = br.readLine()) != null) {
+                out.append(line).append('\n');
+            }
+
+            int exitCode = process.waitFor();
+            if (exitCode != 0) {
+                System.err.println("Error running " + String.join(" ", commands));
+                System.err.println(out.toString());
+            }
+            return exitCode;
+        } catch (Exception ex) {
+            System.err.println("Error running " + String.join(" ", commands));
+            System.err.println(ex.getMessage());
+            return 1;
+        }
+    }
+
+    /**
+     * Compile and load a ddlog program stored in a file.
+     * @param ddlogFile  Pathname to the ddlog program.
+     * @param ddlogInstallationPath  Path to DDLog installation.
+     * @param ddlogLibraryPath  Additional list of paths for needed ddlog libraries.
+     * @return           A DDlogAPI which can be used to access the program.
+     *                   On error returns null and prints an error on stderr.
+     */
+    public static DDlogAPI compileAndLoad(
```

Comment:
Is there a reason we cannot script this? This probably won't work if one wants to set custom flags for one of the commands executed below.

Reply:
What do you mean "script"? You mean "generate a script and run it?"

Reply:
I don't think it needs to be generated, it can be written manually and customized for a particular application such as Weave. Point is, it is probably best if each step in this workflow is handled by a separate program:

1. convert SQL to DDlog
2. compile DDlog code into a dynamic library
3. load and use the library from a Java program.